### PR TITLE
[PTX-23400] Fix VerifyPoolDeleteInvalidPoolID

### PR DESCRIPTION
**Pull request by kphalgun-px**
**What this PR does / why we need it**:
This PR checks whether the error is nil before validating the error message in VerifyPoolDeleteInvalidPoolID. It also filters out nodes on which we cannot delete the pool.

**Which issue(s) this PR fixes** (optional)
Closes #PTX-23400

**Special notes for your reviewer**:
Please review the below Jenkins build for VerifyPoolDeleteInvalidPoolID

https://jenkins.pwx.dev.purestorage.com/job/Users/job/krishna/job/tp-pxe-byoc/2577/

Aetos Dashboard: https://aetos.pwx.purestorage.com/resultSet/testSetID/614947

### CODE REVIEW

**Review by sayalasomayajula-px — APPROVED** [review]
lgtm

**Review by spai-px — APPROVED** [review]
LGTM

**Review by lsrinivas-pure — APPROVED** [review]

File: tests/basic/storage_pool_test.go [hunk 1]
```diff
@@ -6440,8 +6441,21 @@ var _ = Describe("{VerifyPoolDeleteInvalidPoolID}", func() {
...
 		defer appsValidateAndDestroy(contexts)
 
+		kvdbNodes, err := GetAllKvdbNodes()
+		log.FailOnError(err, "failed to get kvdb nodes")
+
+		kvdbNodeIDs := make([]string, 0)
+		for _, kvdbNode := range kvdbNodes {
+			poolIDs, err := GetAllPoolsOnNode(kvdbNode.ID)
+			log.FailOnError(err, "failed to get all pools on node [%s]", kvdbNode.ID)
+			if len(poolIDs) < 2 {
```
[the remark this answers is not in the document]
Comment (kphalgun-px) [reply tests/basic/storage_pool_test.go:6451]: I am assuming a pool can be deleted from a KVDB node if there is more than one; thus, nodes with less than two pools are added to exclusion list.

**Review by spai-px — APPROVED** [review]
LGTM

File: tests/basic/storage_pool_test.go [hunk 2]
```diff
@@ -6440,8 +6441,21 @@ var _ = Describe("{VerifyPoolDeleteInvalidPoolID}", func() {
...
+				log.Infof("Excluding KVDB node [%s] as it has less than 2 pools", kvdbNode.ID)
+				kvdbNodeIDs = append(kvdbNodeIDs, kvdbNode.ID)
```
Comment (sayalasomayajula-px) [inline tests/basic/storage_pool_test.go:6453]: Small doubt, above we are saying excluding the node , but we are including it in to a list of nodes to delete the pools right

Comment (kphalgun-px) [reply tests/basic/storage_pool_test.go:6454]: Here, we are appending the KVDB node with less than two pools to the exclusion list.

**Pull request merged**